### PR TITLE
[RISCV][ISel] Eliminate `andi rd, rs1, -1` instructions

### PR DESCRIPTION
**Pull request by dtcxzyw**
Inspired by https://github.com/llvm/llvm-project/pull/89966, this patch handles the special case `binop_allwusers<and> GPR:$rs1, 0xffffffff -> copy $rs1` to avoid creating redundant `andi rd, rs1, -1` instructions.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/rv64-legal-i32/rv64zba.ll [hunk 1]
```diff
@@ -670,7 +669,6 @@ define i64 @zext_mul21474836480(i32 signext %a) {
 ;
 ; RV64ZBA-LABEL: zext_mul21474836480:
 ; RV64ZBA:       # %bb.0:
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rv64-legal-i32/rv64zba.ll:671]: I guess the FIXME above this function can be removed now?

File: llvm/test/CodeGen/RISCV/rv64-legal-i32/rv64zba.ll [hunk 2]
```diff
@@ -647,7 +647,6 @@ define i64 @zext_mul12884901888(i32 signext %a) {
 ;
 ; RV64ZBA-LABEL: zext_mul12884901888:
 ; RV64ZBA:       # %bb.0:
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rv64-legal-i32/rv64zba.ll:649]: I guess the FIXME above this function can be removed now

File: llvm/test/CodeGen/RISCV/rv64-legal-i32/rv64zba.ll [hunk 3]
```diff
@@ -693,7 +691,6 @@ define i64 @zext_mul38654705664(i32 signext %a) {
 ;
 ; RV64ZBA-LABEL: zext_mul38654705664:
 ; RV64ZBA:       # %bb.0:
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rv64-legal-i32/rv64zba.ll:693]: I guess the FIXME above this function can be removed now?

File: llvm/test/CodeGen/RISCV/rv64zba.ll [hunk 4]
```diff
@@ -864,7 +864,6 @@ define i64 @zext_mul12884901888(i32 signext %a) {
 ;
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/rv64zba.ll:864]: Remove FIXME above this function

**Comment by topperc** [comment]
Some of these look they would have been caught in DAGCombiner if we implement https://github.com/llvm/llvm-project/issues/56645

**Review by topperc — APPROVED** [review]
LGTM

**Review by wangpc-pp — APPROVED** [review]

**Pull request merged**